DesignSafe-CI/portal
# Task/des 2569 add facilities to all project types

## Overview: ##
The overarching goal(DES-2352) is to allow Facilities for all project types as optional and to allow more than one facility. This PR(DES-2569) is a subset that updates the list of facilities, allow facilities for all project types, and makes it optional for all project types. 

## PR Status: ##


## Related Jira tickets: ##

* [DES-2352](https://jira.tacc.utexas.edu/browse/DES-2352)
* [DES-2569](https://jira.tacc.utexas.edu/browse/DES-2569)

## Summary of Changes: ##
1. Update and renamed json file with facilities. (facility-data.json)
2. Add the Facility drop-down to all project types.
3. Add Facility to all modals (manage-experiment, pipeline-primary-ent, pipeline-subentity, published-view, etc)
4. On the backend, add facilities to project types, operations, search_utils, publications, fedora operations

## Testing Steps: ##
For each project type:
1. Make sure the Facility is in the Manage<ProjType> modal with the dropdown. 
2. Selecting a facility saves in the form
3. Using Edit, you can change the facility
4. Work you way down the pipeline, make sure Facility appears
5. When you get to "Proofread <projType>," using  Edit should allow you to change the facility. 
6. Check Published for Facility

Amend:
1. Select your project, go to Amend
2. In Amend, you should be able to change the facility using 'Edit <projType> Metadata' 
3. Check Published for the updated Facility

Published:
1. Select your project, create a new entity with a Facility
2. Version this project. 
3. Check Published for the updated Facility

* For all project types, having NO facility is also an option. Amending to NO facility should update properly. 

* For project type Other, the facility should appear on the project level (but only for Other!)

* For Experimental project type, **Experiment Type & Equipment Type** should appear when the following Facilites are selected: "Large Wave Flume", "Mobile Field Shakers", "Center for Geotechnical Modeling", "Six Degree of Freedom", "Real-Time Multi-Directional", "Boundary Layer Wind Tunnel", "Wall of Wind", "other"

## UI Photos:
<img width="707" alt="Screen Shot 2023-11-08 at 12 40 29 PM" src="https://github.com/DesignSafe-CI/portal/assets/35277477/36db0e75-c991-49df-b82e-72324defb882">
<img width="705" alt="Screen Shot 2023-11-08 at 12 39 28 PM" src="https://github.com/DesignSafe-CI/portal/assets/35277477/d3fbd1f0-a977-44b2-9f22-82d7a8b21a74">
<img width="706" alt="Screen Shot 2023-11-08 at 12 38 44 PM" src="https://github.com/DesignSafe-CI/portal/assets/35277477/2d4618d4-a484-4c86-b407-3a0e0c6fdbb2">
<img width="764" alt="Screen Shot 2023-11-08 at 12 37 22 PM" src="https://github.com/DesignSafe-CI/portal/assets/35277477/639b71cd-8931-48df-8f5b-d92efea4d0d8">
<img width="705" alt="Screen Shot 2023-11-08 at 12 35 59 PM" src="https://github.com/DesignSafe-CI/portal/assets/35277477/89274ecf-2e25-451e-b553-fb1ca31b77e4">
<img width="1042" alt="Screen Shot 2023-11-08 at 12 34 52 PM" src="https://github.com/DesignSafe-CI/portal/assets/35277477/5f08d797-1589-4e00-94b8-e30db986468a">
<img width="1022" alt="Screen Shot 2023-11-08 at 12 33 17 PM" src="https://github.com/DesignSafe-CI/portal/assets/35277477/65de514c-07a1-49a5-b39e-4b4cdc620351">
<img width="713" alt="Screen Shot 2023-11-08 at 12 30 52 PM" src="https://github.com/DesignSafe-CI/portal/assets/35277477/18623da9-3cd2-4215-bb66-3ea879cd8e61">


## Notes: ##
